starknet_py/net/account/base_account.py

```diff
@@ -101,17 +94,35 @@ async def sign_declare_transaction(
         self,
         compiled_contract: str,
         *,
-        compiled_class_hash: Optional[int] = None,
         max_fee: Optional[int] = None,
         auto_estimate: bool = False,
-    ) -> Union[Declare, DeclareV2]:
+    ) -> Declare:
         """
         Create and sign declare transaction.
 
+        :param compiled_contract: string containing a compiled Starknet contract. Supports both new (compiled to sierra)
+            and old cairo contracts.
+        :param max_fee: Max amount of Wei to be paid when executing transaction.
+        :param auto_estimate: Use automatic fee estimation, not recommend as it may lead to high costs.
+        :return: Signed Declare transaction.
+        """
+
+    @abstractmethod
+    async def sign_declare_v2_transaction(
+        self,
+        compiled_contract: str,
+        *,
+        compiled_class_hash,
```

Comment:
I don't think `compiled_class_hash` should still be a keyword argument. Let's move it above `*`

Reply:
Also it should have type defined